cs/playground/ClassCache/Types.cs

```diff
@@ -36,7 +36,7 @@ public class CacheKeySerializer : BinaryObjectSerializer<CacheKey>
     {
         public override void Deserialize(ref CacheKey obj)
```

Comment:
Maybe also change `ref` to `out` to strongly indicate that deserialize needs to provide a value.

Reply:
Good idea, yes that makes sense.

Reply:
We could also symmetrically make `Serialize` take an `in` instead of `ref` object param to signify read-only input. Not sure if it might make it troublesome for some implementations of Serialize, though.  Thoughts?

Reply:
I have updated the PR to use `in` and `out` instead of `ref` for `Serialize` and `Deserialize` respectively.